ksql-engine/src/test/java/io/confluent/ksql/integration/IntegrationTestHarness.java

```diff
@@ -308,6 +310,78 @@ public void produceRecord(final String topicName, final String key, final String
     }
   }
 
+  /**
+   * Verify there are {@code expected} records available to the supplied {@code consumer}.
+   *
+   * @param topic the name of the topic to check.
+   * @param expected the expected rows.
+   * @param valueFormat the format of the value.
+   * @param schema the schema of the value.
+   * @return the list of consumed records.
+   */
+  public List<ConsumerRecord<String, GenericRow>> verifyAvailableRows(
+      final String topic,
+      final Matcher<? super List<ConsumerRecord<String, GenericRow>>> expected,
+      final DataSourceSerDe valueFormat,
+      final Schema schema
+  ) {
+    return verifyAvailableRows(topic, expected, valueFormat, schema, new StringDeserializer());
+  }
+
+  /**
+   * Verify there are {@code expected} records available to the supplied {@code consumer}.
+   *
+   * @param topic the name of the topic to check.
+   * @param expected the expected rows.
+   * @param valueFormat the format of the value.
+   * @param schema the schema of the value.
+   * @param keyDeserializer the keyDeserilizer to use.
+   * @param <K> the type of the key.
+   * @return the list of consumed records.
+   */
+  public <K> List<ConsumerRecord<K, GenericRow>> verifyAvailableRows(
+      final String topic,
+      final Matcher<? super List<ConsumerRecord<K, GenericRow>>> expected,
+      final DataSourceSerDe valueFormat,
+      final Schema schema,
+      final Deserializer<K> keyDeserializer
+  ) {
+    return verifyAvailableRows(
+        topic, expected, valueFormat, schema, keyDeserializer,
+        ConsumerTestUtil.DEFAULT_VERIFY_TIMEOUT);
+  }
+
+  /**
+   * Verify there are {@code expected} records available to the supplied {@code consumer}.
+   *
+   * @param <K> the type of the key.
+   * @param topic the name of the topic to check.
+   * @param expected the expected rows.
+   * @param valueFormat the format of the value.
+   * @param schema the schema of the value.
+   * @param keyDeserializer the key deserilizer to use.
+   * @param timeout the max time to wait for the messages to be received.
+   * @return the list of consumed records.
+   */
+  public <K> List<ConsumerRecord<K, GenericRow>> verifyAvailableRows(
```

Comment:
This adds variants of `verifyAvailableRows' that take a matcher on the list of records.  Allowing us to use this, in combination with `hasUniqueRecords` to wait for a expected map of unique records.